File: src/main/java/fr/lewon/web/bot/entities/input/battle/ActionBattlePlayer.java
package fr.lewon.web.bot.entities.input.battle;

import fr.lewon.bot.http.body.urlencoded.FUEMember;
import fr.lewon.web.bot.entities.battle.BattlePlayer;

public class ActionBattlePlayer extends ActionBattle {

	@FUEMember("who[id_member]")
	private String whoIdMember;

	@FUEMember("who[id_arena]")
	private String whoIdArena;

	public ActionBattlePlayer(BattlePlayer battlePlayer) {
		super(battlePlayer);
		this.whoIdMember = battlePlayer.getIdMember();
		this.whoIdArena = battlePlayer.getIdArena();
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/battle/ActionBattlePlayer.kt
package fr.lewon.bot.hh.entities.input.battle

import com.fasterxml.jackson.annotation.JsonProperty
import fr.lewon.bot.hh.entities.battle.BattlePlayer

class ActionBattlePlayer(battlePlayer: BattlePlayer) : ActionBattle(battlePlayer) {
    @JsonProperty("who[id_member]")
    val whoIdMember: String? = battlePlayer.idMember
    @JsonProperty("who[id_arena]")
    val whoIdArena: String? = battlePlayer.idArena

}